care/facility/api/viewsets/patient_consultation.py

```diff
@@ -17,28 +25,59 @@ class PatientConsultationFilter(filters.FilterSet):
 
 
 class PatientConsultationViewSet(
-    mixins.CreateModelMixin, mixins.ListModelMixin, mixins.RetrieveModelMixin, mixins.UpdateModelMixin, GenericViewSet
+    mixins.CreateModelMixin,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    GenericViewSet,
 ):
     lookup_field = "external_id"
     serializer_class = PatientConsultationSerializer
     permission_classes = (
         IsAuthenticated,
         DRYPermissions,
     )
-    queryset = PatientConsultation.objects.all().select_related("facility").order_by("-id")
+    queryset = (
+        PatientConsultation.objects.all().select_related("facility").order_by("-id")
+    )
     filter_backends = (filters.DjangoFilterBackend,)
     filterset_class = PatientConsultationFilter
 
+    def get_permissions(self):
+        if self.action == "patient_from_asset":
+            return (IsAssetUser(),)
+        return super().get_permissions()
+
     def get_queryset(self):
         if self.request.user.is_superuser:
             return self.queryset
         elif self.request.user.user_type >= User.TYPE_VALUE_MAP["StateLabAdmin"]:
-            return self.queryset.filter(patient__facility__state=self.request.user.state)
+            return self.queryset.filter(
+                patient__facility__state=self.request.user.state
+            )
         elif self.request.user.user_type >= User.TYPE_VALUE_MAP["DistrictLabAdmin"]:
-            return self.queryset.filter(patient__facility__district=self.request.user.district)
+            return self.queryset.filter(
+                patient__facility__district=self.request.user.district
+            )
         allowed_facilities = get_accessible_facilities(self.request.user)
         applied_filters = Q(patient__facility__id__in=allowed_facilities)
         applied_filters |= Q(assigned_to=self.request.user)
         applied_filters |= Q(patient__assigned_to=self.request.user)
         return self.queryset.filter(applied_filters)
 
+    @swagger_auto_schema(
+        responses={200: PatientConsultationIDSerializer},
+    )
+    @action(detail=False, methods=["GET"])
+    def patient_from_asset(self, request):
+        try:
+            consultation = (
+                PatientConsultation.objects.select_related("patient")
+                .order_by("-id")
+                .filter(current_bed__bed__in=request.user.asset.bed_set.all())
+                .only("external_id", "patient__external_id")
+                .first()
```

Comment:
Will there be more than one ?

Reply:
yes, there can be multiple consultation with same current beds as it's an FK.